internal/go_repository.bzl

```diff
@@ -387,8 +388,16 @@ def _generate_package_info(*, importpath, version):
     package_name = importpath
 
     # TODO: Consider adding support for custom remotes.
-    package_url = "https://" + importpath if version else None
+    package_url = "https://{}".format(importpath) if version else None
     package_version = version.removeprefix("v") if version else None
+    # See specification:
+    # https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#golang
+    # scheme:type/namespace/name@version?qualifiers#subpath
+    purl = "pkg:golang/{namespace_and_name}{version}".format(
```

Comment:
We should only set this if `version` is not `None` as if it isn't, we aren't using `go mod` to fetch the module but get it from wherever (say local path or mirror). I would rather not claim any metadata is correct in that case for now.

Reply:
done